### DISCUSSION TITLE
Can I refer to a .csv file not associated with my datasource from within a script? Or treat a .csv file like a table

### DISCUSSION THREAD

**Discussion by parkypoo** [dbeaver/dbeaver#11911] [Q&A] [answered]
Is there a way I can do something like the below scripts I've written in pseudocode?

`SELECT
*
FROM
Customers
WHERE customers.name IN ('C:\Documents\customerlist.csv')`

Or something like
`SELECT
column_1,
column_2
FROM
Customers
UNION SELECT ALL
column_1,
column_2
FROM
'C:\Documents\customerlist.csv'`

I need to cross reference some data with other data sources, and currently I've got to export more data than I need to and then do a lot of processing/filtering in Excel to get the desired results. It's quite cumbersome!

For reference this is using Postgres12 and I only have read privileges on the data, so I can't create any tables to store the data or anything like that.

Any help would be appreciated!

Thank you

**Comment by p91paul** [comment, marked as answer]
Some time ago https://github.com/dbeaver/dbeaver/issues/1804 was requested.
However, doing that is quite difficult and in fact it's only supported by a commercial jdbc driver AFAIK (which is mentioned in that issue).
If you were to buy that driver, configuring it in dbeaver should be tricky due to the lack of dedicated interface but possibly feasible (e.g. playing with driver properties?)

**Reply by parkypoo** [reply]
Thanks Paul,

Do you think a valid alternative could be to create a temporary table, insert the .csv file, then query it from there? I think even with read-only permissions I'm able to do this, but I'm not certain

**Reply by p91paul** [reply]
That would work, but I don't know the first thing about postgres so I have no idea if you are allowed to create temporary tables.
